### PR TITLE
Use reactphp/async instead of clue/reactphp-block

### PR DESCRIPTION
**Pull request by dinooo13**
See https://github.com/reactphp/async

Ref: https://github.com/clue/reactphp-block/pull/67, https://github.com/clue/reactphp-block/issues/68, https://github.com/reactphp/http/pull/464 and https://github.com/clue/reactphp-mq/pull/34

### CODE REVIEW

File: tests/FunctionalTest.php [hunk 1]
```diff
@@ -2,7 +2,6 @@
 
 namespace Clue\Tests\React\HttpProxy;
 
-use Clue\React\Block;
 use Clue\React\HttpProxy\ProxyConnector;
 use React\EventLoop\Loop;
```
Comment (SimonFrings) [inline tests/FunctionalTest.php]: I think we can also get rid of this statement as it's not used anymore in this class

**Review by SimonFrings — COMMENTED** [review]
Just a minor thing I noticed

File: tests/FunctionalTest.php (hunk 1 above)
Comment (SimonFrings) [reply tests/FunctionalTest.php]: I meant:
```php
use React\EventLoop\Loop;
```

Comment (dinooo13) [reply tests/FunctionalTest.php]: Fixed thanks for pointing out 👍🏼

**Review by SimonFrings — APPROVED** [review]
Pull request here looks good, let's get this in :+1:

**Comment by SimonFrings** [comment]
@dinooo13 Just noticed, the test suite is currently not running through as it can be seen in #53 and we need to update our test suite. @yadaiio created a pull request for this in #54. Once this is merged we have to come back to this one here and rebase once. Additionally, we'll probably encounter a conflict in the `composer.json` file when rebasing, as #54 updates the PHPUnit versions which are right below you changes.

I will give an update in here once #54 is merged so we can move on and get this pull request in :+1:

**Comment by SimonFrings** [comment]
@dinooo13 With #54 merged I think we can rebase in here and then get this in :+1:

**Pull request closed without merging**